client/signup/steps/p2-site/index.jsx

```diff
@@ -391,45 +463,86 @@ class P2Site extends Component {
 					className="p2-site__validation-site-title"
 				>
 					<FormLabel htmlFor="site-title-input">
-						{ this.props.translate( 'Name your workspace' ) }
+						{ this.props.translate( 'Workspace name' ) }
 					</FormLabel>
 					<FormTextInput
 						id="site-title-input"
 						autoFocus={ true } // eslint-disable-line jsx-a11y/no-autofocus
 						autoCapitalize={ 'off' }
 						className="p2-site__site-title"
-						disabled={ fieldDisabled }
+						disabled={ submitting }
 						name="site-title"
-						value={ formState.getFieldValue( this.state.form, 'siteTitle' ) }
-						isError={ formState.isFieldInvalid( this.state.form, 'siteTitle' ) }
-						isValid={ formState.isFieldValid( this.state.form, 'siteTitle' ) }
+						value={ siteTitle }
+						isError={ formState.isFieldInvalid( form, 'siteTitle' ) }
+						isValid={ formState.isFieldValid( form, 'siteTitle' ) }
 						onBlur={ this.handleBlur }
 						onChange={ this.handleChangeEvent }
 					/>
+					<FormSettingExplanation className="p2-site__workspace-form-input-explanation">
+						{ this.props.translate( 'This is usually the name of your company or organization' ) }
+					</FormSettingExplanation>
 				</ValidationFieldset>
-				<ValidationFieldset
-					errorMessages={ this.getErrorMessagesWithLogin( 'site' ) }
-					className="p2-site__validation-site"
-				>
-					<FormLabel htmlFor="site-address-input">
-						{ this.props.translate( 'Choose an address for your workspace' ) }
-					</FormLabel>
-					<div className="p2-site__site-url-container">
-						<FormTextInput
-							id="site-address-input"
-							autoCapitalize={ 'off' }
-							className="p2-site__site-url"
-							disabled={ fieldDisabled }
-							name="site"
-							value={ formState.getFieldValue( this.state.form, 'site' ) }
-							isError={ formState.isFieldInvalid( this.state.form, 'site' ) }
-							isValid={ formState.isFieldValid( this.state.form, 'site' ) }
-							onBlur={ this.handleBlur }
-							onChange={ this.handleChangeEvent }
-						/>
-						<span className="p2-site__wordpress-domain-suffix">.wordpress.com</span>
-					</div>
-				</ValidationFieldset>
+				{ showSubdomainInput && (
+					<ValidationFieldset
+						errorMessages={ this.getErrorMessagesWithLogin( 'site' ) }
```

Comment:
Maybe we could split these components into some render functions to improve readability a bit?

Reply:
@unDemian thanks for the suggestion! I've just followed this pattern we're already doing here of having render methods and refactored a bit as you suggested. Probably a further improvement we can try in another pull request is to extract these render methods in explicit React components. What do you think?

Thanks!